lib/screens/transfer/send/send_enter_data/interactor/mappers/send_amount_change_mapper.dart

```diff
@@ -15,11 +15,14 @@ class SendAmountChangeMapper extends StateMapper {
 
     final double currentAvailable = currentState.availableBalance?.amount ?? 0;
 
+    final bool enoughBalance = parsedQuantity <= currentAvailable ||
+        currentState.availableBalance?.asFormattedString() == tokenAmount.asFormattedString();
```

Comment:
Should we be checking the asFormattedString values to find out `enoughBalance` ? 
I would think we would like the check the numbers without any formatting ?

Reply:
This probably warrants a comment but using asFormattedString is a cheap way to take into account precision.

Case
available = 0.01 HUSD
token amount = 0.0103

This only happens when the user enters a fiat currency - then the amount of token can be any double value. 

However they get sent off as 2 digit precision, which amounts to 0.01

0.01 is fine (not too much)
0.0103 would be considered too much even though that's not even shown to the user.

I am sure there's away to do the same sort of rounding or something using the precision int, but ... formatted string is a cheap way to get a 100% accurate reading, since that is the actual number we are actually going to send.

Reply:
String conversion rounds to precision - so would be difficult to accurately re-create. 

Only way I can think of we could do this is by writing our own format as string method, and control the exact way how a double is transformed into a N digit precision string ourselves.

Reply:
Because the double -> string conversion is both shown to the user, and sent on chain - so anything we send, we have also shown to the user, example when the conversion rounds in unexpected ways.